### ISSUE TITLE
add filter to wc_setcookie

### ISSUE THREAD

**Issue by V35CH** [woocommerce/woocommerce#26383]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [X] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [X] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

To compline to cookie law it should be possible to disable cookie storage.
What do you think of a filter in the wc_setcookie function?
Maybe something like this:

```
if(apply_filters( 'woocommerce_cookies_enabled', true )){
setcookie( $name, $value, $expire, COOKIEPATH ? COOKIEPATH : '/', COOKIE_DOMAIN, $secure, apply_filters( 'woocommerce_cookie_httponly', $httponly, $name, $value, $expire, $secure ) );
}
```

**Comment by Konamiman** [comment]
Hi @V35CH, thanks for your suggestion.

If by "cookie law" you mean the GDPR, there are already a number of plugins that allow to display a notice to and act depending on whether the user accepted the cookies or not, for example:

- [Cookie Notice for GDPR & CCPA by dFactory](https://wordpress.org/plugins/cookie-notice)
- [GDPR Cookie Consent Banner by termly](https://wordpress.org/plugins/uk-cookie-consent)
- [EU Cookie Consent Plugin, Wordpress GDPR Compliance by Addify](https://codecanyon.net/item/woocommerce-eu-cookie-consent-plugin-complete-gdpr-law-compliance/21933772)

I'm therefore closing this issue for now. Please don't hesitate to reach to us again if you have further suggestions for improving WooCommerce.

**Comment by V35CH** [comment]
Hello @Konamiman, thank you for your reply and for your plugin suggestions.
But these plugins does only show a cookie banner/notice.
I can not see that these solutions will prevent to set a cookie if the user doesn't opt in to.
So this was my suggestion to add the filter, so I can prevent the system to send a cookie if the user doesn't agreed to receive cookies.

**Comment by dartrax** [comment]
Just a note, this is implemented now since WC 6.2
